### PR TITLE
[DRAFT] Fix bignum const roundtrip

### PR DESCRIPTION
**Pull request by dentiny**
https://github.com/duckdb/duckdb/issues/22027

---

> [!NOTE]
> **Medium Risk**
> Changes core constant-literal parsing so some previously `DOUBLE`-typed large integers now become `BIGNUM`, which can affect query results and type inference for edge-case literals. Risk is mitigated by expanded coverage across `hugeint`/`uhugeint`/`bignum` tests.
> 
> **Overview**
> Large integer literals that overflow `UHUGEINT` are now parsed as `BIGNUM` instead of falling back to `DOUBLE`, preserving exact precision in `Transformer::TransformValue`.
> 
> Tests are updated to expect `BIGNUM` types/values for very large constants, adjust a few bignum casting/aggregation expected results and error messages, and add a new regression test ensuring big numeric literals round-trip losslessly when inserted into `BIGNUM` columns.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 03c82bfd0abdc29d697b7e97eddcfb456156c85d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: test/sql/types/bignum/test_bignum_implicit_cast.test [hunk 1]
```diff
@@ -14,15 +14,15 @@ insert into t values (1), (10), (1000) , (33999999014383402399481480781255147520
 query I
 select sum(a) from t;
 ----
-17976931348623156569122809007601669773980276187098806824731635
+17976931348623157081452776373169450063209456234065280915040278
```
Comment (dentiny) [inline test/sql/types/bignum/test_bignum_implicit_cast.test:17]: If you calculate the sum of these values you will find current existing unit test already suffer precision loss
